Extensions/mutualchecker.js

```diff
@@ -86,12 +86,33 @@ XKit.extensions.mutualchecker = new Object({
 	check: function(json_obj, $link) {
 		if (json_obj.following && !json_obj.customizable) {
 			if (typeof this.mutuals[json_obj.name] === "undefined") {
-				XKit.interface.is_following(json_obj.name, this.preferences.main_blog.value).then(function(follow) {
-					if (follow) {
-						XKit.extensions.mutualchecker.add_label($link, json_obj.name);
-						XKit.extensions.mutualchecker.mutuals[json_obj.name] = true;
-					} else {
-						XKit.extensions.mutualchecker.mutuals[json_obj.name] = false;
+				XKit.tools.Nx_XHR({
+					method: "GET",
+					url: "https://www.tumblr.com/svc/conversations/participant_info?q=" + json_obj.name + "&participant=" + this.preferences.main_blog.value,
+					headers: {
+						"X-Requested-With": "XMLHttpRequest",
+						"X-Tumblr-Form-Key": XKit.interface.form_key()
+					},
+					onerror: function(response) {
+						XKit.extensions.mutualchecker.plain_check(json_obj, $link);
+					},
+					onload: function(response) {
+						try {
+							var raw_data = JSON.parse(response.responseText);
+							if (raw_data.errors) {
+								XKit.extensions.mutualchecker.plain_check(json_obj, $link);
+							} else {
+								var data = raw_data.response;
+								if (data.is_following_blog && data.is_blog_following_you) {
```

Comment:
I get that having the additional information available is good, but now we have two distinct cases for when we'll add the mutual icon. Either we use this, which checks for mutual follow/following, or we use the fallback, in which case we're only checking for if they're following us, and we still have to be careful about where to call this function.

This entangling makes it hard to reason about this code path. We should either 1) drop the is_blog_following_you check here, 2) drop the fallback or 3) add an is_following check to the fallback that's going the other way, if that's possible.

Reply:
mutual checker already only activates on username links with `data-json` that says you're following the blog, so the end result is the same. i just thought that since tumblr provides a double confirmation of it here, it might as well be used